src/control-flow/function-utils.ts

```diff
@@ -0,0 +1,83 @@
+import { Query, type Node as SyntaxNode } from "web-tree-sitter";
+import type { Language } from "./cfg";
+import { extractCFunctionName } from "./cfg-c.ts";
+import { extractCppFunctionName } from "./cfg-cpp.ts";
+import { extractGoFunctionName } from "./cfg-go.ts";
+import { extractPythonFunctionName } from "./cfg-python.ts";
+import { extractTypeScriptFunctionName } from "./cfg-typescript.ts";
+
+/**
+ * Extracts the name of a node by searching for a child node with a specific type.
+ *
+ * @param func - The syntax node to search within.
+ * @param type - The type of the child node to extract the name from.
+ * used among all languages (mostly the easy cases of extracting the name).
+ */
+export function extractNameByNodeType(
+  func: SyntaxNode,
+  type: string,
+): string | undefined {
+  return func.namedChildren.find((child) => child?.type === type)?.text;
+}
+
+/**
+ * Extract a single tagged value from a syntax tree using a Tree-sitter query.
+ *
+ * @param func - The syntax node from which to extract the tree.
+ * @param query - The Tree-sitter query string to execute.
+ * @param tag - The capture tag name to filter by.
+ */
+export function extractTaggedValueFromTreeSitterQuery(
+  func: SyntaxNode,
+  query: string,
+  tag: string,
+): string | undefined {
+  const language = func.tree.language;
+  const queryObj = new Query(language, query);
+
+  const rootNode = func.tree.rootNode;
```

Comment:
I somehow missed this... You're getting the root of the file - this will never work.

Try the following:
- https://github.com/tmr232/function-graph-overview/blob/main/src/control-flow/cfg-python.ts#L359

If you print all the names you get with this, you'll see what I mean.